### PR TITLE
Extract method in Scheduler::Model::Jobs

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Review by Martchus — APPROVED** [review]
I assume you've checked that the test coverage is high enough and also did some manual testing.

**Comment by okurz** [comment]
> I assume you've checked that the test coverage is high enough and also did some manual testing.

I have done neither. But I wanted to create some pull requests to get local changes fully tested in CI and also check if we still run into https://progress.opensuse.org/issues/67855 . Seems like my experiment was successful :D

File: lib/OpenQA/Scheduler/Model/Jobs.pm [hunk 1]
```diff
@@ -34,21 +34,164 @@ use constant MAX_JOB_ALLOCATION => $ENV{OPENQA_SCHEDULER_MAX_JOB_ALLOCATION} //
...
+    try {
+        $worker = $workers->find({id => $worker_id});
+    }
+    catch {
+        log_debug("Failed to retrieve worker ($worker_id) in the DB, reason: $_");    # uncoverable statement
+    };
+    next unless $worker;
+    if ($worker->unfinished_jobs->count) {
+        log_debug "Worker already got jobs, skipping";
+        next;
```
Comment (Amrysliu) [inline lib/OpenQA/Scheduler/Model/Jobs.pm:56]: This `next` is not in a `for` or `foreach`, right? does it work here? The same question for the following `next`.

File: lib/OpenQA/Scheduler/Model/Jobs.pm [hunk 2]
```diff
@@ -34,21 +34,164 @@ use constant MAX_JOB_ALLOCATION => $ENV{OPENQA_SCHEDULER_MAX_JOB_ALLOCATION} //
...
+    }
+    catch {
+        log_debug("Failed to send data to websocket, reason: $_");
+    };
+
+    if (ref($res) eq 'HASH' && $res->{state} && $res->{state}->{msg_sent} == 1) {
+        # note: That only means the websocket server could *start* sending the message but not that the message
+        #       has been received and acknowledged by the worker.
+        log_debug("Sent job(s) '$job_ids_str' to worker '$worker_id'");
+        push(@successfully_allocated, map { {job => $_, worker => $worker_id} } @$job_ids);
```
Comment (coolo) [inline lib/OpenQA/Scheduler/Model/Jobs.pm:157]: this array is not even defined in this function. 

Together with Jane's question: how exactly did you test this?

**Comment by coolo** [comment]
@Martchus gets minus points for bad code review :)

File: lib/OpenQA/Scheduler/Model/Jobs.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/Scheduler/Model/Jobs.pm:157]: I have explained that already in https://github.com/os-autoinst/openQA/pull/3601#issuecomment-735447991

**Review by Martchus — CHANGES_REQUESTED** [review]
> I assume you've checked that the test coverage is high enough and also did some manual testing.

Considering that this wasn't the case I'll request changes until it is. By the way, when testing manually, also make sure you have jobs with all types of dependencies scheduled on your system and check the scheduler log for suspicious messages while that's the case.

**Pull request closed without merging**